pkg/reconciler/reconciler.go

```diff
@@ -0,0 +1,152 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strconv"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/kubeinteraction"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/pipelineascode"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/provider/github"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	pipelinerunreconciler "github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1beta1/pipelinerun"
+	v1beta12 "github.com/tektoncd/pipeline/pkg/client/listers/pipeline/v1beta1"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+)
+
+type Reconciler struct {
+	run               *params.Run
+	pipelineRunLister v1beta12.PipelineRunLister
+	kinteract         *kubeinteraction.Interaction
+}
+
+var _ pipelinerunreconciler.Interface = (*Reconciler)(nil)
+
+var gitAuthSecretAnnotation = "pipelinesascode.tekton.dev/git-auth-secret"
```

Comment:
better use a GroupName  in case they want us to change tekton.dev to something else in the future